File: permutation.py
from os import system


counter = 0


def perm(_list, first, last):
    global counter

    if first == last:

        counter += 1
        return print(f"{counter}. {''.join(_list)}")
    
    for i in range(first, last+1):
        _list[first],_list[i] = _list[i], _list[first]
        perm(_list, first+1, last)
        _list[first],_list[i] = _list[i], _list[first]

    
def main():
    system('cls')
    print('=======(( Start ))=======')

    string = [x for x in input('> Enter Your Array : ')]
    first = 0 
    last  = len(string) - 1
    perm(string, first, last)

    print('=======(( Done ))=======')


if __name__ == '__main__':
    main()

File: permutation.cpp
#include <iostream>
using namespace std;

void perm(string s, const int first, const int last)
{   
    if (first == last) 
        cout<< s << endl;
    
    else
    {
        for (int i = first; i <= last; i++) {
        
            swap(s[first], s[i]);
            perm(s, first+1, last);
            swap(s[first], s[i]);
        }
    }
}

int main()
{
    string str;
    int first = 0;
    int last;

    cout << "=======(( Start ))=======" << endl;
    cout << "> Enter your Array : ";
    cin >> str;

    last = str.length() -1;
    
    perm(str, first, last);

    cout << "=======(( Done )=======";
    return 0;
}
